test/e2e/scenario/internalMonitoring.scenario.ts

```diff
@@ -15,6 +15,7 @@ describe('internal monitoring', () => {
             throw new window.Error('bar')
           },
         }
+        // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
         window.DD_LOGS!.logger.log('hop', context as any)
```

Comment:
not sure to understand what is going on here, do we need the type assertion or not?

Reply:
the rule `@typescript-eslint/no-unnecessary-type-assertion` complains about the `!` in `window.DD_LOGS!.logger`. Removing it makes Typescript complain.

Reply:
which one is right?
why is the other one wrong?

Reply:
Typescript is right, because `window.DD_LOGS` may be `undefined`.
The ESLint rule is wrong, it says that the `!` doesn't change the type of `window.DD_LOGS` which is incorrect. This behavior may be associated with this open issue https://github.com/typescript-eslint/typescript-eslint/issues/2248